app/src/main/java/com/github/weg_li_android/PhotoRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,161 @@
+package com.github.weg_li_android
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.media.ThumbnailUtils
+import android.net.Uri
+import android.util.DisplayMetrics
+import android.util.Size
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.NonNull
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.core.graphics.drawable.toDrawable
+import androidx.databinding.ObservableArrayList
+import androidx.recyclerview.widget.RecyclerView
+import com.github.weg_li_android.ui.main.viewmodel.MainViewModel
+import kotlinx.android.synthetic.main.photo_list_item.view.*
+import timber.log.Timber
+import java.io.FileNotFoundException
+import java.io.IOException
+
+
+class PhotoRecyclerViewAdapter internal constructor(
+    private val context: Context?,
+    private var data: MutableList<Uri>,
+    private val mainViewModel: MainViewModel
+) :
+    RecyclerView.Adapter<PhotoRecyclerViewAdapter.ViewHolder>() {
+
+    private val resources: Resources = context?.resources ?: throw Exception("No context found.")
+    private val mInflater: LayoutInflater = LayoutInflater.from(context)
+    private var mClickListener: ItemClickListener? = null
+    var deletePhotoSet: ObservableArrayList<Int> = ObservableArrayList()
+    /*init {
+        deletePhotoSet.addOnListChangedCallback(
+            RecyclerViewAdapterOnListChangedCallback<ObservableList<Int>>(deletePhotoSet)
+        )
+    }*/
+
+    @NonNull
```

Comment:
Nullable annotations are not needed on Kotlin-only code :)

Reply:
This would also not be needed if we had both java + kotlin code. This is only useful to add on Java code.